test/libsolidity/ASTLegacyJSON.cpp

```diff
@@ -266,16 +266,14 @@ BOOST_AUTO_TEST_CASE(function_type)
 	BOOST_CHECK_EQUAL(argument["attributes"]["name"], "x");
 	BOOST_CHECK_EQUAL(argument["attributes"]["type"], "function () payable external returns (uint256)");
 	Json::Value funType = argument["children"][0];
-	BOOST_CHECK_EQUAL(funType["attributes"]["constant"], false);
```

Comment:
These test are for the legacy JSON (the `true` first parameter). Don't we have tests at all for the new one?!

Reply:
There's a lot more tests in ``ASTLegacyJSON.cpp`` than in ``ASTJSON.cpp`` in general - maybe the idea was to only test cases for which the legacy flag makes a difference in ``ASTJSON.cpp`` and consider the rest covered by ``ASTLegacyJSON.cpp`` already?

That wouldn't have been true for ``constant`` though - but I'm not sure why the new JSON AST used an ``isDeclaredConst`` field there at all...

Reply:
True, `ASTJSON` is for the new one and `ASTLegacyJSON` is for the old one. However it sems the legacy one has twice as many tests.

Reply:
For a lot of cases the AST and the legacy AST are identical and I suspect we didn't want to duplicate the tests. I can move the tests that don't depend on the legacy flag from ``ASTLegacyJSON.cpp`` to ``ASTJSON.cpp`` (this way makes more sense IMHO) and verify that all cases that depend on the legacy flag have two versions and add test cases, if missing.

Reply:
Sounds good.

Reply:
Or maybe I will duplicate all tests, just to be sure.

Reply:
Instead of improving the AST JSON tests here, it may be easier to consider merging #4677 first :)...